### PR TITLE
Fix NumberParser mistakenly returning arab code

### PR DESCRIPTION
**Pull request by snowystinger**
Closes https://github.com/adobe/react-spectrum/issues/3114

## ✅ Pull Request Checklist:

- [ ] Included link to corresponding [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issues).
- [ ] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [ ] Filled out test instructions.
- [ ] Updated documentation (if it already exists for this component).
- [ ] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

### CODE REVIEW

**Review by LFDanLu — DISMISSED** [review]
LGTM, confirmed that the docs and storybook NumberFields don't reproduce this anymore.

File: packages/@internationalized/number/src/NumberParser.ts [hunk 1]
```diff
@@ -215,7 +215,10 @@ function getSymbols(formatter: Intl.NumberFormat, intlOptions: Intl.ResolvedNumb
   let pluralLiterals = allParts.filter(p => !nonLiteralParts.has(p.type)).map(p => escapeRegex(p.value));
   let singularLiterals = singularParts.filter(p => !nonLiteralParts.has(p.type)).map(p => escapeRegex(p.value));
   let sortedLiterals = [...new Set([...singularLiterals, ...pluralLiterals])].sort((a, b) => b.length - a.length);
-  let literals = new RegExp(`${sortedLiterals.join('|')}|[\\p{White_Space}]`, 'gu');
+  let literals = new RegExp('[\\p{White_Space}]', 'gu');
+  if (sortedLiterals.length >= 1) {
+    literals = new RegExp(`${sortedLiterals.join('|')}|[\\p{White_Space}]`, 'gu');
```
Comment (devongovett) [inline packages/@internationalized/number/src/NumberParser.ts:220]: Seems like it's probably compiling an extra regex in the common case (where there are literals). Maybe turn this into a ternary?

Comment (snowystinger) [reply packages/@internationalized/number/src/NumberParser.ts:220]: right, it had a leading `'|'` with nothing preceding it, which wouldn't match anything

**Review by reidbarber — APPROVED** [review]
LGTM

**Review by devongovett — APPROVED** [review]

**Pull request merged**